lib/overcommit/hook/pre_push/base.rb

```diff
@@ -15,6 +16,12 @@ def run?
         (include_remote_ref_deletions? || !@context.remote_ref_deletion?)
     end
 
+    protected
```

Comment:
Why are we using `protected`? There isn't a situation where another instance is attempting to call this method, so `private` is the correct level of access control. If I'm misunderstanding your intent, let me know.

Reply:
same answer as for https://github.com/sds/overcommit/pull/706/files#r386066591